tests/non_literal_attributes.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2018 Guillaume Pinot (@TeXitoi) <texitoi@texitoi.eu>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#[macro_use]
+extern crate structopt;
+
+use std::num::ParseIntError;
+use structopt::clap::AppSettings;
+use structopt::StructOpt;
+
+pub const DISPLAY_ORDER: usize = 2;
+
+// Check if the global settings compile
+#[derive(StructOpt, Debug, PartialEq, Eq)]
+#[structopt(global_settings = &[AppSettings::ColoredHelp])]
+struct Opt {
+    #[structopt(
+        long = "x",
+        display_order = DISPLAY_ORDER,
+        next_line_help = true,
+        default_value = "0",
+        require_equals = true
+    )]
+    x: i32,
+
+    #[structopt(short = "l", long = "level", aliases = &["set-level", "lvl"])]
+    level: String,
+
+    #[structopt(long = "values")]
+    values: Vec<i32>,
+
+    // Note that `raw(...)` is still needed here, for non-convential
+    // methods taking more than one argument.
+    #[structopt(name = "FILE", raw(requires_if = r#""FILE", "values""#))]
```

Comment:
It would be great to support this use case, and remove the raw attribute. As we are on a breaking change version, we can change the syntax. Any idea?

Reply:
One idea is to allow tuples as attributes values and automatically transform them into multiple arguments. As far as I can see, no Clap functions accept tuples, so that should not lead to any conflicts, but theoretically can break later, if Clap decides to introduce a method with a tuple argument on `Arg` or `App`. But that is unlikely.

I mean something like this:

```
#[structopt(name = "FILE", requires_if = ("FILE", "values"))]
```

Reply:
what about supporting `method(arg1, arg2)` with `method = arg1` only a shortcut for `method(arg1)`?

That would be:
```rust
#[structopt(name = "FILE", requires_if("FILE", "values"))]

Reply:
Yes, I think your approach is better.

Reply:
I've implemented your approach and removed any mentions of raw attributes, since they are not needed anymore.